anchor/programs/chack_staking/src/lib.rs

```diff
@@ -0,0 +1,243 @@
+                &ctx.remaining_accounts
+                    .iter()
+                    .map(|account| (account, false, false))
+                    .collect::<Vec<(&AccountInfo, bool, bool)>>(),
+            )
+            .invoke_signed(&[&[
+                ctx.accounts.staking_details.key().as_ref(),
+                &[ctx.bumps["staking_details"]],
+            ]])?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct MplBubblegum;
+
+impl anchor_lang::Id for MplBubblegum {
+    fn id() -> Pubkey {
+        mpl_bubblegum::ID
+    }
+}
+
+#[derive(Accounts)]
+pub struct CreateTree<'info> {
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub tree_config: UncheckedAccount<'info>,
+    /// CHECK: This account must be all zeros.
+    pub merkle_tree: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(
+        seeds = ["tree_owner".as_ref(), merkle_tree.key().as_ref()],
+        bump
+    )]
+    /// CHECK: This account used as a signing PDA only.
+    pub tree_owner: UncheckedAccount<'info>,
+    pub mpl_bubblegum_program: Program<'info, MplBubblegum>,
+}
+
+#[derive(Accounts)]
+pub struct Mint<'info> {
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub tree_config: UncheckedAccount<'info>,
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub merkle_tree: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub owner: Signer<'info>,
+    #[account(
+        seeds = [b"tree_owner", merkle_tree.key().as_ref()],
+        bump
+    )]
+    /// CHECK: This account used as a signing PDA only.
+    pub tree_owner: UncheckedAccount<'info>,
+    pub mpl_bubblegum_program: Program<'info, MplBubblegum>,
+}
+
+#[derive(Accounts)]
+pub struct Stake<'info> {
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub tree_config: UncheckedAccount<'info>,
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub merkle_tree: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub owner: Signer<'info>,
+    #[account(
+        init,
+        seeds = [b"staking_details", merkle_tree.key().as_ref(), owner.key().as_ref()],
+        space = 32 + StakingDetails::INIT_SPACE, // `Pubkey`` + discriminator.
+        payer = owner,
+        bump
+    )]
+    pub staking_details: Account<'info, StakingDetails>,
+    pub mpl_bubblegum_program: Program<'info, MplBubblegum>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct Unstake<'info> {
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub tree_config: UncheckedAccount<'info>,
+    #[account(mut)]
+    /// CHECK: This account is modified in the downstream program.
+    pub merkle_tree: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub owner: Signer<'info>,
+    #[account(
```

Comment:
You're gonna want a close attribute as well so you can stake/unstake more than once. Otherwise init on the Stake IX will fail the second time because the account already exists.